ocm/__main__.py

```diff
@@ -122,17 +122,20 @@ def append(parsed):
 
     component = raw['component']
 
+    if _have_yaml:
+        obj = yaml.safe_load(sys.stdin)
+    else:
+        obj = json.load(sys.stdin)
+
     if parsed.type in ('r', 'resource'):
         attr = component['resources']
     elif parsed.type in ('s', 'source'):
         attr = component['sources']
     elif parsed.type in ('c', 'component-reference'):
         attr = component['componentReferences']
-
-    if _have_yaml:
-        obj = yaml.safe_load(sys.stdin)
-    else:
-        obj = json.load(sys.stdin)
+        for o in obj if isinstance(obj, list) else [obj]:
+            o.setdefault('digest', None)
```

Comment:
oh dear - what have I started. Guess we should switch to using `ocm`'s model-classes 🙈 
background: i initially wanted to keep dependencies to an absolute minimum. then I figured it is a pattern to have an action installing some libs (install-gardener-gha-libs). Ever since then, there is no longer the strong need to avoid using code from cc-utils - also I think I overshot a bit 🙈 

also: do we _have_ to initialse those attrs? Probably, we might also allow absent attrs / define default-nones in ocm/__init__.py, wdyt?

Reply:
agree w.r.t. using `ocm` model classes
didn't consider it as I was not aware of the rational 😅 

---

ocm/init provides sane default values already 🙂